src/event/player/PlayerCreationEvent.php

```diff
@@ -109,4 +117,15 @@ public function setPlayerClass(string $class) : void{
 		Utils::testValidInstance($class, $this->baseClass);
 		$this->playerClass = $class;
 	}
+
+	/**
+	 * Adds a promise to the waiting list for the login sequence.
+	 * Once all the promises that have been added have been completed,
+	 * the player login sequence will be initiated and the player will be created.
+	 *
+	 * @phpstan-param Promise<mixed> $promise
```

Comment:
`Promise<null>` would make more sense, as we won't use any values given by plugins anyway.

(Really it ought to be `Promise<void>`, but I think that would make the signature invalid.)

Reply:
I don't think `void` works properly